Transparency Service configurations and policies (IETF118 hackathon) (#126)
ietf-wg-scitt/draft-ietf-scitt-architecture: draft-ietf-scitt-architecture.md

Each replica MAY provide a recent attestation report for its TEE, binding their hardware platform to the software that runs the Transparency Service, the long-term public key of the service, and the key used by the replica for signing Receipts.
This attestation evidence can be supplemented with Receipts for the software and configuration of the service, as measured in its attestation report.

### Configuration 

The Transparency Service records its configuration in the Append-Only Log using Transparent Statements with distinguished media type `application/x-scitt-configuration`. 

> The registration policy for statements with this media type is implementation-specific. The implementation SHOULD document them, for example defining the Issuers authorized to register configuration Signed Statements. 

Reviewer: ```suggestion
The registration policy for statements with this media type is implementation-specific.
The implementation SHOULD document them, for example defining the Issuers authorized to register configuration Signed Statements. 
```

Each replica MAY provide a recent attestation report for its TEE, binding their hardware platform to the software that runs the Transparency Service, the long-term public key of the service, and the key used by the replica for signing Receipts.
This attestation evidence can be supplemented with Receipts for the software and configuration of the service, as measured in its attestation report.

### Configuration 

> The Transparency Service records its configuration in the Append-Only Log using Transparent Statements with distinguished media type `application/x-scitt-configuration`. 

Reviewer: is this the `3 payload type` in the protected header?
Agree we need a way to determine this. I do wonder how the payload type and mediaType that says it's json, XML, rdf, or others would conflict with this.

The registration policy for statements with this media type is implementation-specific.
The implementation SHOULD document them, for example defining the Issuers authorized to register configuration Signed Statements. 

The Transparency Service is configured by the last Transparent Statement of this type. 

> The Transparency Service MUST register a Signed Statement that defines its initial configuration before registering any other Signed Statement. The Transparency Service MAY register additional Signed Statement that updates its configuration. 

Reviewer: ```suggestion
The Transparency Service MUST register a Signed Statement that defines its initial configuration before registering any other Signed Statement.
The Transparency Service MAY register an additional Signed Statement that updates its configuration. 
```

> The Transparency Service provides an endpoint that returns the Transparency Statement 

Reviewer: ```suggestion
The Transparency Service provides an endpoint that returns the Transparency Statement that describes its current configuration.
```

> that describes its current configuration.

Reviewer: ```suggestion
```

The configuration `reg_info` SHOULD include a secure version number and a timestamp.

The configuration payload uses the CDDL below [TBC].

> ~~~ cddl

Reviewer: ```suggestion

~~~ cddl
```

Each replica MAY provide a recent attestation report for its TEE, binding their hardware platform to the software that runs the Transparency Service, the long-term public key of the service, and the key used by the replica for signing Receipts.
This attestation evidence can be supplemented with Receipts for the software and configuration of the service, as measured in its attestation report.

> ### Configuration

Reviewer 2: We must state something about a default configuration that brings up the `append only log` managed by the Transparency Service. Once the Service is up and running the default configuration can be modified.

Reviewer 2: This comment is ONLY applicable if-we stick to the original definition of the `Configuration` which I strongly believe we should avoid.

Reviewer: I was also struggling with the text that referenced ordering. This sounds implementation-specific. I wonder if we should simply say the Transparency Service MUST have a registration and configuration persisted as signed statements. Should how the service is "booted up" be outside the specification?

Author: The default configuration before registering the first configuration is entirely implementation specific (it just needs to explain how the first explicit configuration is authorized and registered) so I don't think there is much to say about it.

> ### Configuration

Reviewer 2: To make life simple, we should only keep Registration Information as the Element which needs updating, hence using the generic term Configuration, we should adhere to a more specific term known as 
`Registration Policy` which is an updatable element.

Reviewer: This was a great discussion during the hackathon. We talked about having multiple statements for configuration, and registration policies allowing a service to put RBAC on who could read the files. We likely needed an aggregated statement to reference the different config and registration policies. This way the receipt would need only one. This is additional content we wanted to follow-up with after this baseline as it introduces more details we wanted to incrementally think through.

Reviewer: @yogeshbdeshpande, are you ok with deferring the additional detail? Please 👍 if you agree.

Reviewer: @yogeshbdeshpande, please open a new issue/pr if you feel there is missing info.

> The Transparency Service records its configuration in the Append-Only Log using Transparent Statements with distinguished media type `application/scitt-configuration`.

Reviewer 2: ```suggestion
The Transparency Service records its configuration in the Append-Only Log using Signed Statements with distinguished media type `application/scitt-configuration`.
```

Reviewer: This is an interesting question, whether we reference the Signed Statement before or after it's written. 
...records its configuration... could infer the signed statement is being submitted, becoming a transparent statement.
Is there a strong opinion either way?

Author: As usual, what matters for most parties are (signed) transparent statements. In particular, we must reference configuration statements once registered, to ensure they are transparent before they are used for further registrations.

Reviewer: Editors meeting discussion: Signed Statement refers to a configuration that doesn't prove it was placed into the transparency service. 
There's a good discussion whether we should use Transparent Statement or Receipt as the term for referencing something that was registered (notarized). However, we should track that separately. 
For this PR, it's important to recognize the configuration was registered, making "signed statement" too early and incorrect.

> The Transparency Service records its configuration in the Append-Only Log using Transparent Statements with distinguished media type `application/scitt-configuration`.

Author: I agree policy statements are issued by authorized parties, but this seems implementation-specific and I don't think we need to give any details on their authorization here. Hence "Designated authorised entities" seems too strong.

> The Transparency Service records its configuration in the Append-Only Log using Transparent Statements with distinguished media type `application/scitt-configuration`.

Reviewer: Editors call: RBAC is out of scope. Please open a new issue/pr if more discussion is needed.

> The Transparency Service records its configuration in the Append-Only Log using Transparent Statements with distinguished media type `application/x-scitt-configuration`.

Reviewer: ```suggestion
The Transparency Service records its configuration in the Append-Only Log using Transparent Statements with distinguished media type `application/scitt-configuration`.
```

> The registration policy for statements with this media type is implementation-specific.

Reviewer: ```suggestion
The registration policy for statements with the media type suffix (`+<format>` is implementation-specific.
```

The implementation SHOULD document them, for example defining the Issuers authorized to register configuration Signed Statements.

The Transparency Service is configured by the last Transparent Statement of this type.
The Transparency Service MUST register a Signed Statement that defines its initial configuration before registering any other Signed Statement.
The Transparency Service MAY register an additional Signed Statement that updates its configuration.

> The Transparency Service provides an endpoint that returns the Transparency Statement that describes its current configuration.

Reviewer 2: ```suggestion
The Transparency Service provides an endpoint that returns the Transparent Statement that describes its current configuration.
```

> The configuration `reg_info` SHOULD include a secure version number and a timestamp.

Reviewer: To avoid rollbacks, we SHOULD likely make this a MUST, specific to configuration.

Reviewer: Does anyone have an opinion on changing this PR, or making a separate? In the spirt of merging quickly, and iterating, I'm fine deferring to another PR.
